### ISSUE TITLE
Test MariaDB/Aurora with the mysql provider

### ISSUE THREAD

**Issue by SimonCropp** [Particular/NServiceBus.Persistence.Sql#16]
add findings to doco

test with Amazon RDS

**Comment by SimonCropp** [comment]
@gbiellem one for next week?

**Comment by gbiellem** [comment]
Suggest we also verify Amazon Aurora at the same time since it's MySQL compatible as well

**Comment by DavidBoike** [comment]
I tested a reference solution with MariaDB and Amazon Aurora today.

### MariaDB
* AWS default version is 10.0.24, most recent is 10.1.19
* Neither supports JSON datatype, scripts blow up trying to create tables.
* MySQL Workbench app also gives a warning when attempting to connect to a Maria instance that the database isn't 100% compatible, some features will work, others won't, etc.
* Found JSON support is [planned for MariaDB 10.2](https://mariadb.com/kb/en/mariadb/plans-for-10x/)
  * Most recent release on this is [10.2.4 RC](https://mariadb.com/kb/en/mariadb/what-is-mariadb-102/) which was released Feb 17 - one week ago.
  * They use patch numbers for their prereleases, so 10.2.0-1 were alphas, 10.2.2-3 were betas, and 10.2.4 is the first RC.
* Once 10.2 has a finished release, unclear when AWS will support it. Oracle announced [GA of MySQL 5.7](https://www.oracle.com/corporate/pressrelease/mysql-5-7-ga-101915.html) on Oct 19, 2015 and [Amazon didn't announce support for MySQL 5.7](https://aws.amazon.com/about-aws/whats-new/2016/02/amazon-rds-now-supports-mysql-5-7/) until Feb 22, 2016 - 4 months later.
* However, given that Maria JSON support is in RC already, it probably makes sense to just wait until it's finished, or even until RDS offers it. RDS sure makes it easy to test, since you can run MariaDB within the Free Tier.

### Amazon Aurora

* Only one version in AWS, DB param group `default.aurora5.6` which made me suspect from the outset that JSON wouldn't work.
  * And I was right.
* Aurora didn't generate a warning when connecting from MySQL Workbench, however.
* It's **very** unclear, even *murky*, when Amazon will update Aurora to MySQL 5.7 compatibility, although [they said they're working on it](https://forums.aws.amazon.com/thread.jspa?threadID=229475) as of April 15, 2016.
* It seems that if Aurora didn't gain 5.7 compatibility it would become irrelevant, so we should probably just wait on Aurora as well.

**Comment by SimonCropp** [comment]
remaining 10.2 work for mariadb https://jira.mariadb.org/browse/MDEV-11751?jql=project%20%3D%20MDEV%20AND%20issuetype%20%3D%20Task%20AND%20status%20in%20(Open%2C%20Confirmed%2C%20%22In%20Progress%22%2C%20Stalled%2C%20%22In%20Review%22)%20AND%20fixVersion%20in%20(10.2.0%2C%2010.2)%20ORDER%20BY%20priority%20DESC

**Comment by weralabaj** [comment]
Update on MariaDB, there's a new version [10.2.5 RC](https://mariadb.com/kb/en/mariadb/what-is-mariadb-102/).

**Comment by weralabaj** [comment]
Update on Aurora according to [this thread](https://forums.aws.amazon.com/thread.jspa?threadID=229475) "apparently Postgres JSON support is enabled with the Aurora Postgres preview", the preview sing-up is [here](https://pages.awscloud.com/amazon-aurora-with-postgresql-compatibility-preview-form.html).

**Comment by weralabaj** [comment]
Azure MySQL is also using the old version, so no go for us.

**Comment by weralabaj** [comment]
@Particular/sqlpersistence-maintainers Update - MariaDB 10.2.8 is stable branch, so it'll have JSON support https://mariadb.com/kb/en/the-mariadb-library/changes-improvements-in-mariadb-102/. Do we need to test it manually now and update docs? I think that's all that's needed in this issue?

**Comment by DavidBoike** [comment]
Assuming that the same MySQL client works, then I think so. Definitely worth a quick spike to find out. Would we want to change TeamCity to run the MySQL acceptance test suite against a MariaDB server as well before documenting it as supported?

**Comment by weralabaj** [comment]
> Would we want to change TeamCity to run the MySQL acceptance test suite against a MariaDB server as well before documenting it as supported?

Good idea, not sure if we need to run it on every commit though or explicit on every release?

**Comment by MarcinHoppe** [comment]
For starters a smoke test before we release should be good enough, IMO.

**Comment by SimonCropp** [comment]
ok its been two years. i would prefer this no longer showed up as "open issues not fixed" for me. So i am closing. If you are really going to fix it, raise a new issue under someone elses account
